### PR TITLE
chore: upgrade to 1.12.0

### PR DESCRIPTION
**Pull request by chungquantin**
Status: ✅ Tested on my local `devnet`, `testnet` => Block production / finalization + submit extrinsic works well. 

- ISSUE: https://github.com/r0gue-io/pop-node/issues/141
- Release: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.12.0
- Prerequisite: https://github.com/r0gue-io/pop-node/pull/189

### References
- https://github.com/r0gue-io/base-parachain/pull/26
- Compare `pallet-contracts`: [`v1.11.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.10.0/substrate/frame/contracts) and [`v1.12.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.11.0/substrate/frame/contracts)
- Compare `pallet-xcm`: [`v1.11.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.10.0/substrate/frame/xcm) and [`v1.12.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.11.0/substrate/frame/xcm)
## Direct changes
- [[#4329](https://github.com/paritytech/polkadot-sdk/pull/4329)]: Deprecate `NativeElseWasmExecutor` (migrated to `WasmExecutor` in https://github.com/r0gue-io/pop-node/pull/189)
- [[#4295](https://github.com/paritytech/polkadot-sdk/pull/4295)]: Make parachain template async backing ready
- ⚠️ Remove metadata hash (to follow the changes in base-parachain, this will be added back in `1.13.0`)
- [Adds a new configuration item to XCM: XcmRecorder.](https://github.com/paritytech/polkadot-sdk/pull/3872)

### CODE REVIEW

**Comment by peterwht** [comment]
https://github.com/paritytech/polkadot-sdk/compare/polkadot-v1.11.0...polkadot-v1.12.0
```
    from: 0bb6249268c0b77d2834640b84cb52fdd3d7e860
      to: b4016902ac7fc1d885eae236a2f71ddc58abc2f9
 commits: 103
```

# Important commits by [label](https://paritytech.github.io/labels/doc_polkadot-sdk.html)


### T8-polkadot

  (paritytech/polkadot-sdk#4269) Fix polkadot parachains not producing blocks until next session
  (paritytech/polkadot-sdk#4211) Do not re-prepare PVFs if not needed
  (paritytech/polkadot-sdk#4294) rename fragment_tree folder to fragment_chain
  (paritytech/polkadot-sdk#2226) Implementation of the new validator disabling strategy

### T7-smart_contracts

  (paritytech/polkadot-sdk#4213) Contracts: Stabilize XCM host fns

### T6-XCM

  (paritytech/polkadot-sdk#4202) [XCM] Treat recursion limit error as transient in the MQ

### T4-runtime_API

  (paritytech/polkadot-sdk#4301) [Staking] Runtime api if era rewards are pending to be claimed

### T2-pallets

  (paritytech/polkadot-sdk#4202) [XCM] Treat recursion limit error as transient in the MQ
  (paritytech/polkadot-sdk#3865) [balances] Safeguard against consumer ref underflow

### T15-bridges

  (paritytech/polkadot-sdk#4102) Bridge: make some headers submissions free
  (paritytech/polkadot-sdk#4157) Bridge: added free headers submission support to the substrate-relay
  (paritytech/polkadot-sdk#4289) Bump bridges relay version + uncomment bridges zombeinet tests
  (paritytech/polkadot-sdk#4307) Bridges: added helper function to relay single GRANDPA proof + header
  (paritytech/polkadot-sdk#4175) Snowbridge: deposit extra fee to beneficiary on Asset Hub

### T12-benchmarks

  (paritytech/polkadot-sdk#4283) Update approval-voting banchmarks base values
  (paritytech/polkadot-sdk#4285) [subsystem-benchmarks] Log standart deviation for subsystem-benchmarks

### T11-documentation

  (paritytech/polkadot-sdk#4304) fix(seal): shameless fix on sealing typo

### T1-FRAME

  (paritytech/polkadot-sdk#4183) remove unnessisary use statements due to 2021 core prelude

### Credit
Generated with [Tanssi's version-bump-info tool](https://github.com/moondance-labs/tanssi/blob/master/tools/github/print-version-bump-info.ts)

File: runtime/devnet/Cargo.toml [hunk 1]
```diff
@@ -226,16 +225,3 @@ try-runtime = [
...
 experimental = []
-
-# Enable the metadata hash generation.
-#
-# This is hidden behind a feature because it increases the compile time.
-# The wasm binary needs to be compiled twice, once to fetch the metadata,
-# generate the metadata hash and then a second time with the
-# `RUNTIME_METADATA_HASH` environment variable set for the `CheckMetadataHash`
-# extension.
-metadata-hash = ["substrate-wasm-builder/metadata-hash"]
```
Comment (peterwht) [inline runtime/devnet/Cargo.toml:237]: Why was the metadata-hash stuff removed?

**Review by peterwht — APPROVED** [review]

File: runtime/devnet/Cargo.toml (hunk 1 above)
Comment (peterwht) [reply runtime/devnet/Cargo.toml:237]: I see it was added to PR #194

Comment (chungquantin) [reply runtime/devnet/Cargo.toml:237]: Removed to follow the base-parachain upgrade changes.

**Comment by peterwht** [comment]
[sc-1153]

**Review by al3mart — APPROVED** [review]

**Pull request merged**